core/blocks.js

```diff
@@ -14,10 +14,13 @@
  * A mapping of block type names to block prototype objects.
  * @name Blockly.Blocks
  */
-goog.provide('Blockly.Blocks');
+goog.module('Blockly.Blocks');
+goog.module.declareLegacyNamespace();
 
 /**
  * A mapping of block type names to block prototype objects.
  * @type {!Object<string,Object>}
```

Comment:
Could you update the type annotation while you're here? I think it should be `!Object<string, !Object>`. Also I think this needs the [dict](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#dict) annotation since outside devs add properties to it.

Reply:
I've updated the type, but it seems like the `@dict` annotation isn't working when I try to build compressed. 
I'm not going to add the `@dict` for now and I'll see about adding it in a future PR.